src/sw.js

```diff
@@ -0,0 +1,139 @@
+const meta = {
+  name: 'nightTab',
+  version: '7.5.0'
+};
+
+const caching = {};
```

Comment:
Might I suggest that new code that gets added to the codebase uses modern javascript best practices? That way as more people contribute the app is slowly moving forwards instead of stagnating.

What that would look like is not storing everyone on a caching object. Theres no real need for that, when we can define every function as its own function. e.g.
```js
const CACHE_NAME = `${meta.name}-v${meta.version}`
async function cacheFirst(event) {...}
```

Reply:
> All the strategies must be in separate namespace.

In javascript, a file is an isolated [module](https://javascript.info/modules-intro). This is how javascript deals with potential naming conflicts which is the whole point of namespaces. (_since we own all the code, we can rename stuff to have better names too_)

Deeply nesting functions and variables inside of objects is something that was done previously in this codebase, but it's not a good practice.